src/core/libmaven/EIC.cpp

```diff
@@ -146,25 +148,27 @@ void EIC::computeBaseLine(int smoothingWindow, int dropTopX)
     }
     catch (...)
     {
-        cerr << "Exception caught while allocating memory " << n << "floats " << endl;
+        cerr << "Exception caught while allocating memory "
+             << n << " floating point values." << endl;
+        return false;
     }
 
-    if (baselineMode == BaselineMode::Threshold)
-        computeThresholdBaseline(smoothingWindow, dropTopX);
-    else if (baselineMode == BaselineMode::AsLSSmoothing)
-        computeAsLSBaseline(5.0f, 0.05f);
+    return true;
 }
 
-void EIC::computeAsLSBaseline(int lambda, float p, int numIterations)
+void EIC::_computeAsLSBaseline(float lambda, float p, int numIterations)
 {
+    if (!_clearBaseline())
```

Comment:
Can we have this check in `computeBaseLine()` ?  
We won't have to put this check in both the functions then

Reply:
Do you plan to make use of this check in future ? Say, if the baseline could not be cleared, will the user be informed in that case ?

Reply:
Yes this can be included in `computeBaseline()` method itself.

As for you second comment, I think a simple debug message should be fine in such a case?